### ISSUE TITLE
Racer handles raw string incorrectly

### ISSUE THREAD

**Issue by juchiast** [racer-rust/racer#815]
Racer version: 2.0.12

Code snippet:
``` rust
fn main() {
    let s = r#"""#;
    let v = Vec::<u32>::new();
    v.
}
```

Run racer on line 4 charnum 6 gives wrong completion results. This is because the `"` character in the raw string.

PS: This bug is very destructive because Racer cannot complete anything after that raw string.

**Comment by kngwyu** [comment]
I have no idea how I can write a test case...

```rust
#[test]
fn complete_let_after_raw_string() {
    let _lock = sync!();
    let s = r#"
    fn main() {
        let variable = 5;
        let s = r#"""#;
        let a = v~;
        }
    "#;
}
```
Is there any good way?

**Comment by TedDriggs** [comment]
The number of `#` characters you use controls the pattern termination (if you start with 4 `#`, it takes four of them to end the pattern). That'll make it testable; also be sure to test the case where someone uses that pattern in their own code though.

**Comment by kngwyu** [comment]
Thanks.

**Comment by kngwyu** [comment]
Closed via #822
